pkg/networkutils/network.go

```diff
@@ -714,14 +730,38 @@ func listCurrentIptablesRules(ipt iptableswrapper.IPTablesIface, table, chainPre
 	return toClear, nil
 }
 
+func (n *linuxNetwork) cleanOldNatChains(ipt iptableswrapper.IPTablesIface) error {
```

Comment:
Won't this clean up the chains that are already in place? Can we do the following:
1. Clean up any rules with `AWS-SNAT-CHAIN-%d` if they exist
2. Use `AWS-SNAT-CHAIN` for the new rules only and make sure stale rules are cleaned up

Reply:
This would only clean up chains` AWS-SNAT-CHAIN-1` to `AWS-SNAT-CHAIN-<x>` and `AWS-CONNMARK-CHAIN-1` to `AWS-CONNMARK-CHAIN<x>`, which is wanted behavior since the new rules all rely on `-0`. We skip the clean up for a chain if it ends with `-0` and has the SNAT or CONNMARK prefix.

Reply:
Got it, let's add that as a comment above this function. It cleans up rules from previous implementations of this chain

Reply:
We can potentially skip this chain cleanup as the rules are all cleaned up properly, so these existing chains won't affect any iptable rules. Upgrading/downgrading without chain cleanup was tested and there was no breakage. These existing unused chains will be auto-cleaned up during node restart.